code/modules/reagents/newchem/newchem_medicine.dm

```diff
@@ -0,0 +1,805 @@
+				var/obj/item/organ/brain/B = C_target.getorgan(/obj/item/organ/brain)
+				if(B)
+					B.loc = get_turf(C_target)
+					B.transfer_identity(C_target)
+					C_target.internal_organs -= B
+				M.gib(M)
+				return
+		var/mob/dead/observer/ghost = M.get_ghost()
+		M.visible_message("<span class='warning'>[M]'s body convulses a bit.</span>")
+		if(!M.suiciding && !ghost && !(NOCLONE in M.mutations))
+			M.stat = 1
+			M.adjustOxyLoss(-20)
+			M.adjustToxLoss(-20)
+			dead_mob_list -= M
+			living_mob_list |= list(M)
+			M.emote("gasp")
+			add_logs(M, M, "revived", object="strange reagent")
+			hardset_dna(M, null, null, null, null, /datum/species/zombie)
+	..()
+	return
+*/
+datum/reagent/strange_reagent/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(prob(rand(1,100)))
+		M.adjustBruteLoss(2*REM)
+		M.adjustFireLoss(2*REM)
+	..()
+	return
+
+/datum/chemical_reaction/strange_reagent
+	name = "Strange Reagent"
+	id = "strange_reagent"
+	result = "strange_reagent"
+	required_reagents = list("omnizine" = 1, "holywater" = 1, "mutagen" = 1)
+	result_amount = 3
+
+datum/reagent/life
+	name = "Life"
+	id = "life"
+	description = ""
+	reagent_state = LIQUID
+	color = "#C8A5DC" // rgb: 200, 165, 220
+	custom_metabolism = 0.2
+
+/datum/chemical_reaction/life
+	name = "Life"
+	id = "life"
+	result = "life"
+	required_reagents = list("strange_reagent" = 1, "synthflesh" = 1, "blood" = 1)
+	result_amount = 3
+	required_temp = 374
+
+/datum/chemical_reaction/life/on_reaction(var/datum/reagents/holder, var/created_volume)
+	chemical_mob_spawn(holder, 1, "Life")
+
+proc/chemical_mob_spawn(var/datum/reagents/holder, var/amount_to_spawn, var/reaction_name)
+	if(holder && holder.my_atom)
+		var/blocked = list(/mob/living/simple_animal/hostile,
+			/mob/living/simple_animal/hostile/pirate,
+			/mob/living/simple_animal/hostile/pirate/ranged,
+			/mob/living/simple_animal/hostile/russian,
+			/mob/living/simple_animal/hostile/russian/ranged,
+			/mob/living/simple_animal/hostile/syndicate,
+			/mob/living/simple_animal/hostile/syndicate/melee,
+			/mob/living/simple_animal/hostile/syndicate/melee/space,
+			/mob/living/simple_animal/hostile/syndicate/ranged,
+			/mob/living/simple_animal/hostile/syndicate/ranged/space,
+			/mob/living/simple_animal/hostile/alien/queen/large,
+			/mob/living/simple_animal/hostile/retaliate,
+			/mob/living/simple_animal/hostile/retaliate/clown,
+			/mob/living/simple_animal/hostile/mushroom,
+			/mob/living/simple_animal/hostile/asteroid,
+			/mob/living/simple_animal/hostile/asteroid/basilisk,
+			/mob/living/simple_animal/hostile/asteroid/goldgrub,
+			/mob/living/simple_animal/hostile/asteroid/goliath,
+			/mob/living/simple_animal/hostile/asteroid/hivelord,
+			/mob/living/simple_animal/hostile/asteroid/hivelordbrood,
+			/mob/living/simple_animal/hostile/carp/holocarp,
+			/mob/living/simple_animal/hostile/mining_drone
+			)//exclusion list for things you don't want the reaction to create.
```

Comment:
Still, it lists all the asteroid mobs instead of just hostile/asteroid. Is it using type comparison or istype()?